src/components/IntlMessage.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { FormattedMessage, MessageValue } from "react-intl";
+
+interface BaseProps {
+  id: string;
+  values?: { [key: string]: MessageValue | JSX.Element };
+}
+
+interface RenderByTagNameProps {
+  className?: string;
+  tagName?: keyof JSX.IntrinsicElements;
+}
+
+type CustomRendererProps =
+  | { onlyRenderText: boolean }
+  | { render: (message: string[]) => React.ReactNode };
+
+type Props =
+  | BaseProps
+  | (BaseProps & (RenderByTagNameProps | CustomRendererProps));
+
+/**
+ * Neat component for rendering internationalized text in a variety of ways.
+ *
+ * react-intl's FormattedMessage class does everything we need, but the syntax
+ * for rendering in various ways is not always straightforward.
```

Comment:
Could you add an example of how to use this component to this comment?

Reply:
8c2ec18